test/FunctionalTests/Microsoft.OData.Client.Tests/Serialization/MockHttpClientHandler.cs

```diff
@@ -0,0 +1,88 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="MockHttpClientHandler.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.OData.Client.Tests.Serialization
+{
+    /// <summary>
+    /// A mock implementation of <see cref="HttpClientHandler"/>
+    /// for testing purposes.
+    /// </summary>
+    internal sealed class MockHttpClientHandler : HttpClientHandler
+    {
+        private readonly Func<HttpRequestMessage, HttpResponseMessage> _requestHandler;
+        private readonly List<string> _requests = new List<string>();
+
+        
+
+        /// <summary>
+        /// Creates an instance of <see cref="MockHttpClientHandler"/> that
+        /// returns the specified response contents for any request.
+        /// </summary>
+        /// <param name="expectedResponse">The contents to return in each response.</param>
+        public MockHttpClientHandler(string expectedResponse)
+        {
+            _requestHandler = (_) =>
+            {
+                var response = new HttpResponseMessage(HttpStatusCode.OK);
+                response.Content = new StringContent(expectedResponse);
+                return response;
+            };

```

Comment:
```suggestion
        : this((_) =>
            {
                var response = new HttpResponseMessage(HttpStatusCode.OK);
                response.Content = new StringContent(expectedResponse);
                return response;
            })
        {
```

I'm not sure if this actually compiles, or that it's even readable, but I think there's something to be said about the explicit nature of having a primary constructor.

Reply:
It does compile.